### PR TITLE
[Delivers #91153332] fix rendering of error pages

### PR DESCRIPTION
**Pull request by afeld**
screen shot 2015-04-02 at 9 46 18 pm

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class HomeController < ApplicationController
   before_filter :setup_mygov_client
   before_filter :setup_mygov_access_token
+  before_action :authenticate_user!, only: :error
```
Comment (afeld) [inline app/controllers/home_controller.rb]: Just so we don't get spam.

File: config/application.rb [hunk 2]
```diff
@@ -41,7 +41,6 @@ class Application < Rails::Application
     }
     config.roadie.url_options = config.action_mailer.default_url_options
 
-    config.exceptions_app = self.routes
```
Comment (afeld) [inline config/application.rb]: This is the actual fix.

File: app/controllers/communicarts_controller.rb [hunk 3]
```diff
@@ -78,7 +78,7 @@ def user_id
 
   def authentication_error(e)
     flash[:error] = e.message
-    redirect_to "/498.html"
+    render 'authentication_error', status: 498
```
Comment (cmc333333) [inline app/controllers/communicarts_controller.rb]: 498? That looks to be specific to ESRI. Should this be a 403?

File: config/routes.rb [hunk 4]
```diff
@@ -3,6 +3,7 @@
   match 'approval_response', to: 'communicarts#approval_response', via: [:get, :put]
   root :to => 'home#index'
   match "/auth/:provider/callback" => "home#oauth_callback", via: [:get]
+  get '/error' => 'home#error'
```
Comment (cmc333333) [inline config/routes.rb]: Is this only supposed to be accessible when in dev mode?

**Comment by cmc333333** [comment]
There's a lot of parts I don't understand here; can you explain why this fixes?

File: app/controllers/communicarts_controller.rb (hunk 3 above)
Comment (afeld) [inline app/controllers/communicarts_controller.rb]: Heh, I didn't even question it. Good catch.

File: config/routes.rb (hunk 4 above)
Comment (afeld) [inline config/routes.rb]: No, I think it's useful in various environments for testing things like New Relic and logging.

**Comment by afeld** [comment]
> There's a lot of parts I don't understand here

Take a look at the commits separately – might make more sense?

File: config/routes.rb (hunk 4 above)
Comment (afeld) [inline config/routes.rb]: ...and that the 500 page shows up properly :smile:

**Comment by cmc333333** [comment]
Should have looked up `config.exceptions_app`, now I think I understand, thanks.

**Pull request merged**